i18n(es): Translate internationalization.mdx

ansango opened the pull request:
#### Description (required)

Translate internationalization.mdx into -es

astrobot-houston commented:
Hello! Thank you for opening your **first PR** to Astro’s Docs! 🎉

Here’s what will happen next:

1. Our GitHub bots will run to check your changes.
   If they spot any broken links you will see some error messages on this PR.
   Don’t hesitate to ask any questions if you’re not sure what these mean!

2. In a few minutes, you’ll be able to see a preview of your changes on Vercel 🥳

3. One or more of our maintainers will take a look and may ask you to make changes.
   We try to be responsive, but don’t worry if this takes a few days.

astrobot-houston commented:
<!-- lunaria-action-comment -->

## Lunaria Status Overview


🌕 **This pull request will trigger status changes.**

<details>
<summary>Learn more</summary>

By default, every PR changing files present in the [Lunaria configuration's `files` property](https://lunaria.dev/reference/configuration/#files-required) will be considered and trigger status changes accordingly.

You can change this by adding one of the keywords present in the [`ignoreKeywords` property in your Lunaria configuration file](https://lunaria.dev/reference/configuration/#ignorekeywords) in the PR's title (ignoring all files) or by [including a tracker directive](https://lunaria.dev/guides/tracking/#tracker-directives) in the merged commit's description. 
</details>

### Tracked Files


| Locale | File                                                                                                                                                                             | Note                                             |
| ------ | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------ |
| es     | [guides/internationalization.mdx](https://github.com/withastro/docs/blob/f77ef12fc94eeafd5663c898c5f062f919519ccd/src%2Fcontent%2Fdocs%2Fes%2Fguides%2Finternationalization.mdx) | Localization added, will be marked as complete.  |

<details>
<summary>Warnings reference</summary>

| Icon | Description                                                                                                                                       |
| ---- | ------------------------------------------------------------------------------------------------------------------------------------------------- |
| 🔄️  | The source for this localization has been updated since the creation of this pull request, make sure all changes in the source have been applied. |

</details>

ansango commented on src/content/docs/es/guides/internationalization.mdx:
```diff
@@ -0,0 +1,374 @@
+
+La internacionalización (i18n) de Astro te permite adaptar tu proyecto para una audiencia internacional. Esta API de enrutamiento te ayuda a generar, usar y verificar las URL que produce tu sitio multilingüe.
+
+Astro's i18n routing allows you to bring your multilingual content with support for configuring a default language, computing relative page URLs, and accepting preferred languages provided by your visitor's browser. You can also specify fallback languages on a per-language basis so that your visitors can always be directed to existing content on your site.
```
Listo 😄

Waxer59 commented on src/content/docs/es/guides/internationalization.mdx:
```diff
@@ -0,0 +1,372 @@
+---
+title: Internationalización (i18n) Enrutamiento
```
```suggestion
title: Enrutamiento con Internationalización (i18n)
```

Waxer59 commented on src/content/docs/es/guides/internationalization.mdx:
```diff
@@ -0,0 +1,372 @@
+
+Astro utiliza un [middleware](/es/guides/middleware/) para implementar su lógica de enrutamiento. Esta función de middleware se coloca en la [primera posición](/es/guides/middleware/#encadenando-el-middleware) donde espera cada `Response` que proviene de cualquier middleware adicional y de cada ruta antes de ejecutar finalmente su propia lógica.
+
+Estro significa que las operaciones (por ejemplo, redirecciones) de tu propio middleware y tu lógica de página se ejecutan primero, tus rutas se renderizan y luego el middleware de i18n realiza sus propias acciones, como verificar que una URL localizada corresponda a una ruta válida.
```
```suggestion
Esto significa que las operaciones (por ejemplo, redirecciones) de tu propio middleware y tu lógica de página se ejecutan primero, tus rutas se renderizan y luego el middleware de i18n realiza sus propias acciones, como verificar que una URL localizada corresponda a una ruta válida.
```

Waxer59 commented on src/content/docs/es/guides/internationalization.mdx:
```diff
@@ -0,0 +1,372 @@
+
+### Crear enlaces
+
+With i18n routing configured, you can now compute links to pages within your site using the helper functions such as [`getRelativeLocaleURL()`](/en/reference/api-reference/#getrelativelocaleurl) available from the [`astro:i18n` module](/en/reference/api-reference/#internationalization-astroi18n). These generated links will always provide the correct, localized route and can help you correctly use ...
```
```suggestion
```

Waxer59 commented on src/content/docs/es/guides/internationalization.mdx:
```diff
@@ -0,0 +1,372 @@
+---
+import { getRelativeLocaleUrl } from 'astro:i18n';
+
+// defaultLocale is "es"
```
```suggestion
// defaultLocale es "es"
```

Waxer59 commented on src/content/docs/es/guides/internationalization.mdx:
````diff
@@ -0,0 +1,372 @@
+
+```js title="src/middleware.js"
+import { defineMiddleware } from "astro:middleware";
+import { redirectToDefaultLocale } from "astro:i18n"; // función disponible con `manual` routing
````
```suggestion
import { redirectToDefaultLocale } from "astro:i18n"; // función disponible con enrutamiento `manual`
```

Waxer59 commented on src/content/docs/es/guides/internationalization.mdx:
````diff
@@ -0,0 +1,372 @@
+})
+```
+
+Cuando uses funciones del módulo virtual [`astro:i18n`](/es/reference/api-reference/#internacionalización-astroi18n) para calcular rutas de URL válidas basadas en tu configuración (por ejemplo, `getRelativeLocaleUrl()`), [usa el `path` como valor para `locale`](/es/reference/api-reference/#getlocalebypath).
````
```suggestion
Cuando uses funciones del [módulo virtual `astro:i18n`](/es/reference/api-reference/#internacionalización-astroi18n) para calcular rutas de URL válidas basadas en tu configuración (por ejemplo, `getRelativeLocaleUrl()`), [usa el `path` como valor para `locale`](/es/reference/api-reference/#getlocalebypath).
```

Waxer59 commented on src/content/docs/es/guides/internationalization.mdx:
```diff
@@ -0,0 +1,372 @@
+
+Para que coincidan correctamente con las preferencias de tus usuarios, proporciona tus `codes` utilizando el mismo patrón [utilizado por el navegador](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language#syntax).
+
+## Estrategia de respaldo o Fallback
```
```suggestion
## Estrategia de respaldo
```

Waxer59 commented on src/content/docs/es/guides/internationalization.mdx:
```diff
@@ -0,0 +1,372 @@
+Esta función tiene algunas restricciones:
+
+- La opción `site` es obligatoria.
+- El `output` debe ser `server` para que esta función funcione.
```
```suggestion
- El `output` debe ser `server`.
```

Waxer59 commented on src/content/docs/es/guides/internationalization.mdx:
```diff
@@ -0,0 +1,372 @@
+Astro depende de los siguientes encabezados para admitir la función:
+
+- [`X-Forwarded-Host`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Host) y [`Host`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host). Astro usará lo principal primero. Si no existe, intentará lo secundario.
+- [`X-Forwarded-Proto`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto) y [`URL#protocol`](https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol) de la solicitud.
```
```suggestion
- [`X-Forwarded-Proto`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto) y [`URL#protocol`](https://developer.mozilla.org/en-US/docs/Web/API/URL/protocol) de la solicitud del servidor.
```

Waxer59 reviewed (commented):
Buena PR 🤠, te dejo unas sugerencias!

dreyfus92 commented on src/content/docs/es/guides/internationalization.mdx:
```diff
@@ -0,0 +1,371 @@
+- La opción `site` es obligatoria.
+- El `output` debe ser `server`.
+- No puede haber páginas prerenderizadas individuales.
+- The adapter feature [`functionPerRoute`](/en/reference/adapter-reference/#functionperroute) is not supported.
```
```suggestion
```

dreyfus92 reviewed (commented):
Buen trabajo @ansango 😁, te dejo una pequeña sugerencia.

ansango commented:
> Buen trabajo @ansango 😁, te dejo una pequeña sugerencia.

Gracias, 8 ojos ven más que dos! 😅

dreyfus92 reviewed (approved):
Muchas gracias @ansango y bienvenido a `astro-docs` 🫡

dreyfus92 commented:
@ansango, parece que todavía queda pendiente resolver los links que están mal. Si necesitas ayuda me puedes pinguear aquí o en discord.

ansango commented:
@dreyfus92 Creo que ya deberia de funcionar todo correctamente, links arreglados y probados 😄

yanthomasdev commented:
Looks like we're missing a few headings in the Spanish translation of API Reference (e.g. [Astro.preferredLocale](https://docs.astro.build/en/reference/api-reference/#astropreferredlocale)), that's why the broken links. Could any of you take a look and make sure to add these to the `api-reference.mdx` translation?

ansango commented:
> Looks like we're missing a few headings in the Spanish translation of API Reference (e.g. [Astro.preferredLocale](https://docs.astro.build/en/reference/api-reference/#astropreferredlocale)), that's why the broken links. Could any of you take a look and make sure to add these to the `api-reference.mdx` translation?

Thank you for the review @yanthomasdev, I fixed and added new translations in `api-reference.mdx`. I hope build will be ok! 😄

dreyfus92 commented:
There are still 4 broken links, we're almost there to ship this @ansango 😁

ansango commented:
> There are still 4 broken links, we're almost there to ship this @ansango 😁

Sure! I reviewed the previous gh action, and I built on my desktop, so I hope that we'll be ready now 😄

dreyfus92 commented:
Muchas gracias @ansango, ya quedó 🥳
